crates/polars-ops/src/series/ops/arg_min_max.rs

```diff
@@ -224,45 +249,35 @@ where
     for<'b> &'b [T::Native]: ArgMinMax,
 {
     match ca.is_sorted_flag() {
-        IsSorted::Ascending => Some(0),
-        IsSorted::Descending => Some(ca.len() - 1),
+        IsSorted::Ascending => ca.first_non_null(),
+        IsSorted::Descending => ca.last_non_null(),
         IsSorted::Not => {
             ca.downcast_iter()
                 .fold((None, None, 0), |acc, arr| {
                     if arr.len() == 0 {
                         return acc;
                     }
-                    let chunk_min_idx: Option<usize>;
-                    let chunk_min_val: Option<T::Native>;
-                    if arr.null_count() > 0 {
-                        // When there are nulls, we should compare Option<T::Native>
-                        chunk_min_val = None; // because None < Some(_)
-                        chunk_min_idx = arr
-                            .into_iter()
+                    let chunk_min: Option<(usize, T::Native)> = if arr.null_count() > 0 {
+                        arr.into_iter()
                             .enumerate()
+                            .flat_map(|(idx, val)| val.map(|val| (idx, *val)))
```

Comment:
The previous implementation involved many comparisons of `Options<T>`, and it is important to pay great attention to `None < Some(_)`. I filtered out the `None` value directly here. I'm not very sure if this will have a negative impact, but the previous logic was a bit too complicated.

Reply:
Can we rewrite this one in `downcast_iter` as well?

Reply:
Sorry, I didn't quite understand what you meant. Could you please explain it in more detail? Thanks!

Reply:
Oh sorry, my bad. This one is good. :+1:  (missed the context due to github collapsing)